src/components/UNSTABLE__Shell/HeaderAction/index.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file Header action.
+ * @copyright IBM Security 2020
+ */
+
+import classnames from 'classnames';
+import { bool, func, node, string } from 'prop-types';
+import React, { cloneElement, useRef } from 'react';
+
+import IconButton from '../../IconButton';
+import Transition from '../../Transition';
+
+import { namespace } from '../Shell';
+import { headerNamespace } from '../ShellHeader';
+
+import popoverNamespace from '../Popover';
+
+function HeaderAction({
+  activeAction,
+  children,
+  className,
+  hasBadge,
+  id,
+  popover,
+  setActiveAction,
+  ...other
+}) {
+  const isActive = popover && activeAction === id;
+  const ref = useRef(null);
+
+  const headerButtonNamespace = `${headerNamespace}__button`;
+  const isIconButton = popover && children.type === IconButton;
+
+  return (
+    <>
+      {cloneElement(children, {
+        className: classnames(className, {
+          [`${namespace}__button--icon`]: isIconButton,
+          [headerButtonNamespace]: isIconButton,
+          [`${headerButtonNamespace}--notifications`]: isIconButton && hasBadge,
+          [`${headerButtonNamespace}--active`]: isIconButton && isActive,
+        }),
+        onClick: () => {
+          setActiveAction(!isActive ? id : null);
+
+          if (!isActive) {
+            ref.current.focus();
+          }
+        },
```

Comment:
`onBlur` will have to be added here to handle when the action loses focus. 
```
        onBlur: () => isActive && setActiveAction(null),
```
This worked for me when running the storybook.

Reply:
Good catch!

Reply:
@SimonFinney that is right in line with what I was thinking! But when I run it locally and select and account's dropdown it still removes the popover:

<img width="374" alt="Screen Shot 2020-02-18 at 12 09 03 PM" src="https://user-images.githubusercontent.com/7417858/74759878-7a865100-5247-11ea-9e67-1f34e57056fb.png">